src/js/select2/data/select.js

```diff
@@ -114,7 +114,12 @@ define([
     });
 
     container.on('unselect', function (params) {
-      self.unselect(params.data);
+      container.$element.find('option').each(function () {
```

Comment:
@kevin-brown I'm feeling a bit uncertain about whether this change is a good idea, after looking at it again.

This extra `for` loop feels like it could quite easily introduce performance regressions.

I'll find time to take another look at this soon, but if in any doubt, please feel free to revert this change.

Reply:
(update: re-inspecting this changeset currently)

Reply:
Yep, a slightly stronger statement after re-reading the changeset and related issues: I think I'd prefer for this pull request to be reverted (and #6128 re-opened), if that's OK.

I didn't add any test coverage as part of the changes, which was unwise, because that makes it much harder to determine whether the modifications truly resolve the issue identified in the linked ticket.

Reply:
(revert pull request opened as #6217)